File: Core/Entities/OrderAggregate/Order.cs
namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        // Regular properties
        public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.UtcNow;
        public required string BuyerEmail { get; set; }

        public Guid DeliveryMethodId { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; } = null!;

        // Owned entities
        public ShippingAddress ShippingAddress { get; set; } = null!;
        public PaymentSummary PaymentSummary { get; set; } = null!;

        // Regular properties
        public decimal Subtotal { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public required string PaymentIntentId { get; set; }

        // Navigation property to related entities
        public IReadOnlyList<OrderItem> OrderItems { get; set; } = new List<OrderItem>();

        public decimal GetTotal()
        {
            return Subtotal + DeliveryMethod.Price;
        }
    }
}

File: Client/src/app/shared/models/order.ts
import { DeliveryMethod } from "./deliveryMethod"

export interface Order {
  id: string
  orderDate: string
  buyerEmail: string
  deliveryMethodId: string
  deliveryMethod: DeliveryMethod
  shippingAddress: ShippingAddress
  paymentSummary: PaymentSummary
  subtotal: number
  discount?: number
  status: number
  statusString: string
  paymentIntentId: string
  orderItems: OrderItem[]
  total: number
}

// export interface DeliveryMethod {
//   id: string
//   shortName: string
//   deliveryTime: string
//   description: string
//   price: number
// }

export interface ShippingAddress {
  name: string
  line1: string
  line2: any
  city: string
  state: string
  postalCode: string
  country: string
}

export interface PaymentSummary {
  last4: number
  brand: string
  expirationMonth: number
  expirationYear: number
}

export interface OrderItem {
  id: string
  itemOrdered: ItemOrdered
  price: number
  quantity: number
}

export interface ItemOrdered {
  productId: string
  productName: string
  pictureUrl: string
}

export interface OrderToCreate {
  shoppingCartId: string;
  deliveryMethodId: string;
  shippingAddress: ShippingAddress;
  paymentSummary: PaymentSummary;
  discount?: number
}